app/Http/Controllers/PostsController.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Post;
+use App\PostStatus;
+use App\User;
+use Illuminate\Http\Request;
+
+class PostsController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+
+        $posts = Post::all();
+
+        return view('posts.index')->with('posts', $posts);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        //
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param int $id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $post                      = Post::find($id);
+        $post_attributes           = $post->getAttributes();
+        $post_attributes['author'] = $post->user->name;
+        unset($post_attributes['author_id']);
+        $post_attributes['status'] = $post->status->status;
+        unset($post_attributes['status_id']);
+
+        return view('posts.details')->with('post', $post_attributes);
+    }
+
+    /**
+     * Show the form for editing the specified resource.
+     *
+     * @param int $id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function edit($id)
+    {
+        //
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param int $id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $post = Post::find($id);
+        $post->update($request->all());
+        $post->status_id = $request->status_id;
+        $post->author_id = $request->author_id;
+        $post->save();
+
+        return view('posts.updated')->with('post', $post);
+    }
+
+    /**
+     * Display the edit form with the predefined fields.
+     *
+     * @param int $id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function updateDataFromView($id)
+    {
+        $users         = User::all()->toArray();
```

Comment:
You do not need to convert Collection to array (at least not here).
Please Revise this chapter
https://laravel.com/docs/5.8/eloquent-collections
> All collections also serve as iterators, allowing you to loop over them as if they were simple PHP arrays

Reply:
Done.